File: tests/fixtures/javascript/clean/secrets.js
// Clean fixture: secrets-detection rules should NOT trigger here

// API key from environment — compliant
const API_KEY = process.env.API_KEY;

// AWS credentials from environment — compliant
const AWS_ACCESS_KEY = process.env.AWS_ACCESS_KEY_ID;
const AWS_SECRET_KEY = process.env.AWS_SECRET_ACCESS_KEY;

// GitHub token from environment — compliant
const GITHUB_TOKEN = process.env.GITHUB_TOKEN;

// DB URL from environment — compliant
const DB_URL = process.env.DATABASE_URL;

// JWT secret from environment — compliant
const JWT_SECRET = process.env.JWT_SECRET;

async function fetchData() {
  const apiKey = process.env.API_KEY;
  const res = await fetch("https://api.example.com", {
    headers: { Authorization: `Bearer ${apiKey}` },
  });
  return res.json();
}

// Short placeholder strings should not trigger
const placeholder = "changeme";
const testValue = "test";

File: tests/fixtures/python/clean/secrets.py
# Clean fixture: secrets-detection rules should NOT trigger here

import os

import requests


def get_api_key():
    """Load API key from environment — compliant."""
    return os.environ["API_KEY"]


def get_aws_credentials():
    """Load AWS credentials from environment — compliant."""
    access_key = os.getenv("AWS_ACCESS_KEY_ID")
    secret_key = os.environ.get("AWS_SECRET_ACCESS_KEY")
    return access_key, secret_key


def get_github_token():
    """Token loaded from env — compliant."""
    return os.getenv("GITHUB_TOKEN", "")


def get_database_url():
    """DB URL from environment — compliant."""
    return os.environ["DATABASE_URL"]


def get_jwt_secret():
    """JWT secret from env — compliant."""
    return os.getenv("JWT_SECRET")


def connect_to_api():
    """Uses env-sourced credentials — compliant."""
    api_key = os.getenv("API_KEY")
    headers = {"Authorization": f"Bearer {api_key}"}
    return requests.get("https://api.example.com", headers=headers)


# These string literals should not trigger (too short, not matching patterns)
placeholder = "changeme"
test_value = "test"
